File: CarHomeWidget/src/main/java/com/anod/car/home/skin/icon/BBBIconProcessor.java
package com.anod.car.home.skin.icon;

import android.graphics.Bitmap;

public class BBBIconProcessor implements IconProcessor {

	private static final float SIZE_DIFF = .3f;

	@Override
	public Bitmap process(Bitmap icon) {
		int x = (int)(icon.getWidth() * SIZE_DIFF);
		int w = icon.getWidth() - x;
		int y = (int)(icon.getHeight() * SIZE_DIFF);
		int h = icon.getHeight() - y;

		return Bitmap.createBitmap(icon, x, 0, w, h);
	}

	@Override
	public float getSizeDiff() {
		return SIZE_DIFF;
	}

	@Override
	public String id() {
		return "bbb";
	}

}

File: app/src/main/java/com/anod/car/home/skin/icon/BBBIconProcessor.kt
package com.anod.car.home.skin.icon

import android.graphics.Bitmap

class BBBIconProcessor : IconProcessor {

    override fun process(icon: Bitmap): Bitmap {
        val x = (icon.width * sizeDiff).toInt()
        val w = icon.width - x
        val y = (icon.height * sizeDiff).toInt()
        val h = icon.height - y

        return Bitmap.createBitmap(icon, x, 0, w, h)
    }

    override val sizeDiff = .3f
    override val id = "bbb"

}
